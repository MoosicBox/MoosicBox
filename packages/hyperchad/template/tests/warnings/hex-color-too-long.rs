//! Test case for hex colors that are too long (more than 8 characters)
//! 
//! Valid hex colors must be exactly 3, 6, or 8 hexadecimal digits.
//! This test tries a 9-character hex color which should be rejected.
//!
//! Pattern: #1e2abcdef
//! Tokenizes as: # + LitFloat(1e2, suffix="abcdef")
//! Total length: "1e2" (3) + "abcdef" (6) = 9 characters (invalid)

use hyperchad_template::container;

fn main() {
    // This should fail with: "Invalid hex color '#1e2abcdef'. 
    // Hex colors must be 3, 6, or 8 hexadecimal digits"
    let _ = container! {
        div color=#1e2abcdef { "9 characters - too long" }
    };
}
