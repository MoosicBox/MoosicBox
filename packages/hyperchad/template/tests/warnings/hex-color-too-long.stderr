error: Invalid hex color '#1e2abcdef'. Hex colors must be 3, 6, or 8 hexadecimal digits (0-9, a-f)
  --> tests/warnings/hex-color-too-long.rs:16:19
   |
16 |         div color=#1e2abcdef { "9 characters - too long" }
   |                   ^^^^^^^^^^
